cpp/src/io/json/parser_features.cpp

```diff
@@ -99,10 +107,22 @@ struct empty_column_functor {
       child_columns.push_back(cudf::type_dispatcher(
         schema.child_types.at(child_name).type, *this, schema.child_types.at(child_name)));
     }
-    return make_structs_column(0, std::move(child_columns), 0, {}, stream, mr);
+    return std::make_unique<column>(cudf::data_type{type_id::STRUCT},
+                                    0,
+                                    rmm::device_buffer{},
+                                    rmm::device_buffer{},
+                                    0,
+                                    std::move(child_columns));
```

Comment:
What does this do different than the factory call?

Reply:
We discovered through profiling that `make_lists_column` and `make_structs_column` introduce significant overhead by unnecessarily calling to `purge_non_empty_nulls` and `superimpose_nulls_no_sanitize`. These sanitization may be needed in general, however, here we know that the child(ren) of JSON output column cannot have non-null rows if the parent is null (due to invalid input) thus we want to avoid such overhead by calling to the column constructor directly.

Reply:
@karthikeyann Please add comments to clarify why:
```
// Do not use `cudf::make_lists_column` since we do not need to call `purge_nonempty_nulls`
// on the child column as it does not have non-empty nulls.
```
and
```
// Do not use `cudf::make_structs_column` since we do not need to call `superimpose_nulls`
// on the children columns.
```

Reply:
I've filed a related issue for this: https://github.com/NVIDIA/spark-rapids-jni/issues/2602

Reply:
https://github.com/rapidsai/cudf/issues/12567

Reply:
Seems we could fix the `make_structs_column` and `make_lists_column` to not call these utilities for empty columns. 
Or the utilities could be fixed to no-op for empty columns perhaps?

Reply:
Either fix is good.

Reply:
Ah yes, I was wrong on this: it is unnecessary here since the input column is empty.